Code/GraphMol/Substruct/SubstructUtils.cpp

```diff
@@ -219,18 +219,18 @@ void removeDuplicates(std::vector<MatchVectType> &matches,
   //  that the 4 paths are equivalent in the semantics of the query.
   //  Also, OELib returns the same results
   //
-  std::set<boost::dynamic_bitset<>> seen;
+  std::unordered_set<boost::dynamic_bitset<>> seen;
```

Comment:
would a `std::unordered_set<std::string>` work here since `std::string` instances are mutable?  A small benchmark showed this result with gcc -O3
```
[std::unordered_set<std::string>]
Insertion time: 52 ms
Lookup time: 12 ms

[std::unordered_set<boost::dynamic_bitset>]
Insertion time: 152 ms
Lookup time: 405 ms
```

Reply:
Thanks for that idea @whosayn . I just did a quick bit of benchmarking and there is probably a small improvement when using `std::string` instead of `boost::dynamic_bitset`

I will update the PR